### PR TITLE
REFACTOR-#6116: Move `groupby_ohlc` implementation to QC layer

### LINKED ISSUE #6116: REFACTOR: add query compiler interface for groupby.ohlc
upstreaming https://github.com/ponder-org/modin-public/commit/b7fd64cabff8c881e70bb00416571ca85d93aae6

### PR DESCRIPTION
**Pull request by RehanSD**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [ ] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [ ] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [ ] signed commit with `git commit -s` 
- [ ] Resolves #6116 ? 
- [ ] tests added and passing
- [ ] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

**Review by mvashishtha — COMMENTED** [review]
@RehanSD could you please add tests for groupby ohlc?

**Comment by RehanSD** [comment]
Necessary PR in Soda: https://github.com/ponder-org/soda/pull/533

File: modin/core/storage_formats/base/query_compiler.py [hunk 1]
```diff
@@ -3452,6 +3452,17 @@ def groupby_unique(
...
+    def groupby_ohlc(
+        self,
+        by,
+        axis,
+        groupby_kwargs,
+        agg_args,
+        agg_kwargs,
+        is_df,
+    ):
+        return DataFrameDefault.register(lambda df: df.ohlc())(self)
```
Comment (vnlitvinov) [inline modin/core/storage_formats/base/query_compiler.py:3464]: this isn't using any of arguments to the function... why is that?

**Review by vnlitvinov — COMMENTED** [review]
@RehanSD this PR is failing CI, pls fix

File: modin/core/storage_formats/base/query_compiler.py (hunk 1 above)
Comment (RehanSD) [reply modin/core/storage_formats/base/query_compiler.py:3464]: When defaulting to pandas, we don't need any of these arguments. These are for other storage formats that won't default to pandas!

**Review by naren-ponder — APPROVED** [review]

**Review by vnlitvinov — APPROVED** [review]
LGTM!

**Pull request merged**